### PR TITLE
[Sync] Allocmem and Freemem conversion

### PR DESCRIPTION
**Pull request by kiranchandramohan**
List of patches:
https://reviews.llvm.org/D114104 : Convert allocmem and freemem fir operations to malloc and free. @AlexisPerry 
https://reviews.llvm.org/D115797 : Add test with shape for the allocmem and freemem ops

### CODE REVIEW

**Comment by AlexisPerry** [comment]
Assuming the tests pass this looks good to me, but please wait for one other approver to give the go-ahead before merging.

**Review by schweitzpgi — APPROVED** [review]
Thanks for synching this back.

**Pull request merged**
